packages/bns/src/index.ts

```diff
@@ -0,0 +1,748 @@
+    contractName: BNS_CONTRACT_NAME,
+    functionName: options.functionName,
+    senderAddress: options.senderAddress,
+    functionArgs: options.functionArgs,
+    network: options.network,
+  });
+}
+
+/**
+ * Check if name can be registered
+ *
+ * @param {string} fullyQualifiedName - the fully qualified name to check
+ * @param {StacksNetwork} network - the Stacks network to broadcast transaction to
+ *
+ * @returns {Promise} that resolves to true if the operation succeeds
+ */
+export async function canRegisterName(
+  fullyQualifiedName: string,
+  network: StacksNetwork
+): Promise<boolean> {
+  const bnsFunctionName = 'can-name-be-registered';
+  const { subdomain, namespace, name } = decodeFQN(fullyQualifiedName);
+  if (subdomain) {
+    throw new Error('Cannot register a subdomain using registerName');
+  }
+
+  // Create a random address as input to read-only function call
+  // Not used by BNS contract function but required by core node API
+  // https://github.com/blockstack/stacks-blockchain/blob/master/src/net/http.rs#L1796
+  const randomPrivateKey = privateKeyToString(makeRandomPrivKey());
+  const randomAddress = getAddressFromPrivateKey(randomPrivateKey);
+
+  return callReadOnlyBnsFunction({
+    functionName: bnsFunctionName,
+    senderAddress: randomAddress,
+    functionArgs: [bufferCVFromString(namespace), bufferCVFromString(name)],
+    network,
+  }).then((responseCV: ClarityValue) => {
+    if (responseCV.type === ClarityType.ResponseOk) {
+      return responseCV.value.type === ClarityType.BoolTrue;
+    } else {
+      return false;
+    }
+  });
+}
+
+/**
+ * Get price of namespace registration in microstacks
+ *
+ * @param {string} namespace - the namespace
+ * @param {StacksNetwork} network - the Stacks network to use
+ *
+ * @returns {Promise} that resolves to a BN object number of microstacks if the operation succeeds
+ */
+export async function getNamespacePrice(namespace: string, network: StacksNetwork): Promise<BN> {
+  const bnsFunctionName = 'get-namespace-price';
+
+  // Create a random address as input to read-only function call
+  // Not used by BNS contract function but required by core node API
+  // https://github.com/blockstack/stacks-blockchain/blob/master/src/net/http.rs#L1796
+  const randomPrivateKey = privateKeyToString(makeRandomPrivKey());
+  const randomAddress = getAddressFromPrivateKey(randomPrivateKey);
+
+  return callReadOnlyBnsFunction({
+    functionName: bnsFunctionName,
+    senderAddress: randomAddress,
+    functionArgs: [bufferCVFromString(namespace)],
+    network,
+  }).then((responseCV: ClarityValue) => {
+    if (responseCV.type === ClarityType.ResponseOk) {
+      if (responseCV.value.type === ClarityType.Int || responseCV.value.type === ClarityType.UInt) {
+        return responseCV.value.value;
+      } else {
+        throw new Error('Response did not contain a number');
+      }
+    } else {
+      const errorResponse = responseCV as ResponseErrorCV;
```

Comment:
Would be a bit safer/friendlier in these Clarity decodings to check for the `ResponseError` type here in the `else if (...)`, then have the `else` throw an error for unexpected Clarity value